src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -848,6 +847,26 @@ private static void SetKestrelUrlOverrideEnvVariables(this IResourceBuilder<Proj
         });
     }
 
+    private static string ParseKestrelHost(string host)
+    {
+        if (string.Equals(host, "localhost", StringComparison.OrdinalIgnoreCase))
+        {
+            // If the host is localhost, we set it to null so that it uses the default host
+            return "localhost";
+        }
+        else if (IPAddress.TryParse(host, out var _))
+        {
+            // If the host is an IP address, we use it as is
+            return host;
+        }
+
+        // This is a simplified version of the Asp.NET logic for parsing host URLs
+        // If the host is not localhost or an IP address, it's treated as binding to all
+        // interfaces. In this case, we assume the IPv4 0.0.0.0 address rather than testing
+        // for IPv6 vs. IPv4 support.
+        return "0.0.0.0";
```

Comment:
I would be a bit happier if we detected if we are IPv6-only machine and use `::` in that case. Granted, these are not super common.

Reply:
I did actually start taking that approach; the question becomes what level of IPv6 support? I can look for non-loopback IPv6 addresses, but at least on Mac there's a private network IPv6 address allocated for a bridge100 interface regardless. We can do the detection, but it's not always a great user experience.

Reply:
But now that I actually read, you're right, we can fall back to :: if IPv4 isn't actually supported.

Reply:
Updated to ensure IPv4 support before selecting 0.0.0.0.